libs/common/result.hpp

```diff
@@ -114,13 +125,45 @@ namespace iroha {
      * result
      */
     template <typename T, typename E, typename Transform>
-    constexpr auto operator|(Result<T, E> r, Transform &&f)
-        -> decltype(f(std::declval<T>())) {
+    constexpr auto operator|(Result<T, E> r, Transform &&f) ->
+        typename std::enable_if<
```

Comment:
Update doc plz

Reply:
Will be doen

Reply:
Actually, docs need not be updated because changes just resolve overloads for  different types of functions.